hybrasyl/Messaging/ChatCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+                    var argtext = (string)x.GetField("ArgumentText", BindingFlags.Public | BindingFlags.Static).GetValue(null);
+                    var options = argtext.Split('|');
+                    var allowedArgcounts = new List<int>();
+                        
+                    foreach (var option in options)
+                    {
+                        // Count number of required / optional arguments
+                        var split = ArgsRegex.Split(option);
+                        if (split.Count() == 1)
+                            allowedArgcounts.Add(option.Count(e => e == '<'));
+                        else
+                        {
+                            // Now get the bracket contents
+
+                            foreach (Group group in ArgsRegex.Match(argtext).Groups)
+                            {
+                                var baseopt = argtext.Remove(argtext.IndexOf(group.Value)).Count(e => e == '<');
+                                allowedArgcounts.Add(baseopt + group.Value.Count(e => e == '<'));
+                                allowedArgcounts.Add(baseopt);
+                            }
+                        }
+
+                    }
+                    // int argcount = ((string)x.GetField("ArgumentText", BindingFlags.Public | BindingFlags.Static).GetValue(null)).Count(e => e == '<');
+                    _associates.Add(command, (x, allowedArgcounts));
+                    Logger.Info($"{command} registered");
+                }
+                catch (Exception e)
+                {
+                    Logger.Warn($"{x.Name}: could not be loaded - {e}");
+                }
+            }
+        }
+
+        public void IsHandler(string command) => _associates.ContainsKey(command);
+
+        public bool TryGetHandler(string command, out Type handler)
+        {
+            handler = null;
+            if (_associates.ContainsKey(command))
+            {
+                handler = _associates[command].Type;
+                return true;
+            }
+            return false;
+        }
+
+        public void Handle(User user, string command, string args)
+        {
+            if (_associates.ContainsKey(command))
+            {
+                var strings = QuotesRegex.Matches(args).Cast<Match>().Select(m => m.Value).ToArray();
+
+                var handler = _associates[command];
+                var priv = (bool) handler.Type.GetField("Privileged", BindingFlags.Public | BindingFlags.Static).GetValue(null);
+
+                if (priv && !user.IsPrivileged)
+                {
+                    user.SendSystemMessage("Failed: Access denied (command is privileged)");
+                    UserLogger.Error($"{user.Name}: denied attempt to use privileged command {command}");
+                    return;
+                }
+
+                var splitArgs = QuotesRegex.Split(args).Select(e => e.Replace("\"", "")).ToArray();
```

Comment:
For commands with no additional args, this statement appears to result in a single arg of "", which then causes it to enter the evaluation below (essentially failing). Based on tests of /gcm /reload /debug and /help

Reply:
fixed